pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/default.nix

```
@@ -0,0 +1,56 @@
+{ mkDerivation, lib, fetchFromGitHub, pkgconfig, python3, cmake, ninja
+, qtbase, qtimageformats, enchant, xdg_utils, desktop-file-utils, ffmpeg
+, openalSoft, lzma, lz4, xxHash, zlib, minizip, openssl, libtgvoip
+, rlottie-tdesktop, range-v3
+}:
+
+with lib;
+
+mkDerivation rec {
+  pname = "kotatogram-desktop";
+  version = "1.1.4";
+
+  src = fetchFromGitHub {
+    owner = "kotatogram";
+    repo = "kotatogram-desktop";
+    rev = "k${version}";
+    sha256 = "0ys593dddv3v86fnzglk68c3wmdhz1ypcgnqjfzd3idq68pmh6r8";
+    fetchSubmodules = true;
+  };
+
+  postPatch = ''
+    substituteInPlace Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp \
+      --replace '"libenchant-2.so.2"' '"${enchant}/lib/libenchant-2.so.2"'
+  '';
+
+  nativeBuildInputs = [ pkgconfig python3 cmake ninja ];
+
+  buildInputs = [
+    qtbase qtimageformats ffmpeg openalSoft lzma lz4 xxHash
+    zlib minizip openssl enchant libtgvoip rlottie-tdesktop range-v3
+  ];
+
+  enableParallelBuilding = true;
+
+  qtWrapperArgs = [
+    "--prefix PATH : ${xdg_utils}/bin"
+    "--prefix PATH : ${desktop-file-utils}/bin"
+  ];
+
+  cmakeFlags = [
+    "-DTDESKTOP_API_TEST=ON"
+    "-DTDESKTOP_DISABLE_GTK_INTEGRATION=ON"
+  ];
+
+  meta = {
+    description = "Kotatogram – experimental Telegram Desktop fork";
+    longDescription = ''
+      Unofficial desktop client for the Telegram messenger, based on Telegram Desktop.
+      It contains some useful (or purely cosmetic) features, but they could be unstable. A detailed list is available here: https://kotatogram.github.io/changes
+    '';
+    license = licenses.gpl3;
+    platforms = platforms.linux;
```

if you don't want to troubleshoot aarch issues
```suggestion
    platforms = platforms.linux;
    badPlatforms = platforms.aarch64;
```

---

issue seems to be an upstream issue:
```
/build/source/test/view/set_symmetric_difference.cpp:324:1: internal compiler error: in output_constructor_regular_field, at varasm.c:5207
  324 | }
      | ^
```

---

> if you don't want to troubleshoot aarch issues

But issue is in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/range-v3/default.nix
This can be fixed if maintainer of range-v3 disablse tests for aarch64

---

Right, I'll disable it in https://github.com/NixOS/nixpkgs/pull/78447 (was aware of it but initially hoping that it'll be fixed soon, but I didn't find an upstream fix and don't have an AArch64 test setup atm - theoretically it should work though, at least the Alpine Linux package builds).

---

Why just not to disable doCheck for aarch64?
The library is header-only and errors are from tests

---

if you can verify that it's only the tests which are affected, then that's fine. Just worried about the instance where the build is actually affected.

---

> Why just not to disable doCheck for aarch64?
The library is header-only and errors are from tests

I wasn't sure it the errors are actually from the tests (didn't have a closer look and I'm used that the tests run after the actual build - like with Makefiles or Python). Let's see if it works, if not I'll just disable it until someone can properly test and fix it.

And please add further comments on the other PR if we don't add the commit in this PR.